components/inputs/input-text.js

```diff
@@ -332,6 +354,11 @@ class InputText extends FormElementMixin(SkeletonMixin(RtlMixin(LitElement))) {
 				this.requestValidate(false);
 				this.setFormValue(this.value);
 				this._prevValue = (oldVal === undefined) ? '' : oldVal;
+			} else if (prop === 'unit') {
+				if (this.unit && this.unit !== '%') {
```

Comment:
As discussed, when/if we support other units, we will either translate them internally for the label or expose another property allowing the consumer to specify.  For "%", the symbol is announced correctly by screen readers.